pkg/network/ebpf/c/protocols/postgres/decoding.h

```diff
@@ -226,33 +246,47 @@ static __always_inline bool handle_response(pktbuf_t pkt, conn_tuple_t conn_tupl
     }
 
 #pragma unroll(POSTGRES_MAX_MESSAGES_PER_TAIL_CALL)
-    for (__u32 iteration = 0; iteration < POSTGRES_MAX_MESSAGES_PER_TAIL_CALL; ++iteration) {
-        if (!read_message_header(pkt, &header)) {
+    for (__u8 messages_count = 0; messages_count < POSTGRES_MAX_MESSAGES_PER_TAIL_CALL; ++messages_count) {
+        read_result = read_message_header(pkt, &header);
+        if (read_result != READ_OK) {
             break;
         }
         if (header.message_tag == POSTGRES_COMMAND_COMPLETE_MAGIC_BYTE) {
-            handle_command_complete(&conn_tuple, transaction);
             found_command_complete = true;
             break;
         }
         // We didn't find a command complete message, so we advance the data offset to the end of the message.
         // reminder, the message length includes the size of the payload, 4 bytes of the message length itself, but not
         // the message tag. So we need to add 1 to the message length to jump over the entire message.
         pktbuf_advance(pkt, header.message_len + 1);
+        iteration_value->msg_count++;
     }
 
+    if (found_command_complete || (read_result == READ_FRAGMENTED)) {
+        // if completion was found or the packet was fragmented, then report the transaction and clear it from the map.
```

Comment:
Handling command complete if we're dealing with a fragmented packet is wrong and must be reverted

It damages our reported latency per message and makes the metric unreliable

Reply:
removed condition of fragmented packet